### PR TITLE
Flow-366: c2 : list envelope documents

### PR DESCRIPTION
**Pull request by harshitav-docusign**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/DocuSignDemo/apiDefinition.swagger.json [hunk 1]
```diff
@@ -2115,6 +2193,59 @@
...
       }
     },
+    "/accounts/{accountId}/envelopes/{envelopeId}/envelopeDocuments": {
+      "get": {
+        "tags": [
+          "DocuSign"
+        ],
+        "summary": "List envelope documents",
+        "description": "List documents in an envelope.",
+        "operationId": "GetEnvelopeDocumentsInfo",
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSignDemo/apiDefinition.swagger.json:2203]: maybe you could simply name  this as `ListEnvelopeDocuments`

File: certified-connectors/DocuSignDemo/script.csx [hunk 2]
```diff
@@ -747,6 +747,15 @@ public class Script : ScriptBase
     var uriLogicAppsBase64 = Convert.ToBase64String(Encoding.UTF8.GetBytes(uriLogicApps ?? string.Empty));
     var notificationProxyUri = this.Context.CreateNotificationUri($"/webhook_response?logicAppsUri={uriLogicAppsBase64}");
 
+    // TODO: This map is added for backward compatibility. This will be removed once old events are deprecated
+	var envelopeEventMap = new Dictionary<string, string>() {
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSignDemo/script.csx:751]: fix indent

**Comment by harshitav-docusign** [comment]
I will create one more PR pointing to dev branch. This one I pointed to release branch.

File: certified-connectors/DocuSignDemo/script.csx [hunk 3]
```diff
@@ -1614,6 +1642,134 @@ public class Script : ScriptBase
...
+
+      if (body["envelopeDocuments"] == null)
+      {
+        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Envelope do not contain documents");
+      }
+
+      foreach(var document in body["envelopeDocuments"] as JArray ?? new JArray())
+      {
+        envelopeDocuments.Add(new JObject(
+          new JProperty("documentId", document["documentId"]),
```
Comment (sriharip-docusign) [inline certified-connectors/DocuSignDemo/script.csx:1764]: You can use shorthand.

```javascript
=new JObject()
        {
          ["documentId"] = document["workflowId"],
          ["name"] = document["defaultName"]
        };
```

**Comment by sriharip-docusign** [comment]
Ok. Please address comments from these PR when creating new one.
Also I notice there were merge conflicts. So you might want to resync your fork and resolve the conflicts.

**Pull request closed without merging**